File: 01_gpio_digital/led_blink_task.c
#include <wiringPi.h>
#define LED_PIN1 0
#define LED_PIN2 2
#define LED_PIN3 3


int main(void)
{
    wiringPiSetup ();
    //wiringPiSetupGpio();
    pinMode (LED_PIN1, OUTPUT);
    pinMode (LED_PIN2, OUTPUT);
    pinMode (LED_PIN3, OUTPUT);

    digitalWrite(LED_PIN1, HIGH); 
    delay(2000);
    digitalWrite(LED_PIN1, LOW); 
    digitalWrite(LED_PIN2, HIGH); 
    delay(2000);
    digitalWrite(LED_PIN2, LOW); 
    digitalWrite(LED_PIN3, HIGH); 
    delay(2000);
    digitalWrite(LED_PIN3, LOW); 
    delay(2000);


    
    return 0;



}

File: 01_gpio_digital/led_blink_task.py
import RPi.GPIO as GPIO
import time

LED_PIN1 = 22
GPIO.setmode(GPIO.BCM) #GPIO.BCM or GPIO.BOARD
GPIO.setup(LED_PIN1, GPIO.OUT) #GPIO.IN
LED_PIN2 = 27
GPIO.setup(LED_PIN2, GPIO.OUT) #GPIO.IN
LED_PIN3 = 17
GPIO.setup(LED_PIN3, GPIO.OUT) #GPIO.IN

    
GPIO.output(LED_PIN1, GPIO.HIGH) #1
print("led on")
time.sleep(2)
GPIO.output(LED_PIN1, GPIO.LOW) #0 
print("led off")

GPIO.output(LED_PIN2, GPIO.HIGH) #1
print("led on")
time.sleep(2)
GPIO.output(LED_PIN2, GPIO.LOW) #0 
print("led off")

GPIO.output(LED_PIN3, GPIO.HIGH) #1
print("led on")
time.sleep(2)
GPIO.output(LED_PIN3, GPIO.LOW) #0 
print("led off")

GPIO.cleanup() #GPIO 핀상태 초기화